src/pages/Search/Search.js

```diff
@@ -56,10 +56,26 @@ function Search({ setData, setActiveCell }) {
 
 
   useEffect(() => {
+    function sortData(items) {
```

Comment:
I would move this logic to the reducer. It fits better there imo.

Reply:
I can't find a practical example about it to follow. Everywhere I find reducers should stay pure and with no logic and avoid side effects. Do you mean maybe with a middleware or thunk or so?

Reply:
[According to the docs](https://redux.js.org/faq/code-structure#how-should-i-split-my-logic-between-reducers-and-action-creators-where-should-my-business-logic-go) there are some advantages to putting business logic inside the actions vs reducers. You can pick either one. But I wouldn't put it in the component. In the end it's also a matter of taste I guess.

Reply:
As for a practical example: you can simply move the complete sortData function to the reducer file. Then just call it like this (don't know the names of your variables so it's more of an example)

```
function reducer(state, action) {
  switch(action.type) {
    case 'POSTS_FETCHED":
      return {
        ...state,
        posts: sortData(action.data)
      }
  }
}
```

Note: I wrote this on my phone and it's been a while since I used redux. so no guarantee that this works correctly :smile:

Reply:
Also think about the naming of the function sortPosts. Is it really sorting posts? I would expect an array to be returned that is sorted in a specific order with array.sort. IMO naming is very important (as you may have seen in my reviews :smile:). This is not sorting but grouping. And to be more specific grouping by day and hour. So a good name for this function would be `groupPostsByWeekdayAndHour`. I know naming is a bit difficult in the beginning but try to stop for a moment and think about what a function does or variable contains. Get really specific and select a name accordingly. It's ok to pick long names